### PR TITLE
stable/node-local-dns Extend daemonset arg list

### PR DESCRIPTION
**Pull request by kkonnov**
## Description

Extend daemonset arg list

## Checklist

- [ ] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)
- [ ] I have read the [contribution instructions](https://github.com/deliveryhero/helm-charts#opening-a-pr), bumped chart version and regenerated the docs
- [ ] Github actions are passing

### CODE REVIEW

File: stable/node-local-dns/values.yaml [hunk 1]
```diff
@@ -19,6 +19,18 @@ config:
   # Port used for the health endpoint
   healthPort: 8080
 
+  setupInterface: {}
+
+  setupIptables: {}
+
+  skipTeardown: {}
```
Comment (max-rocket-internet) [inline stable/node-local-dns/values.yaml:26]: I think these are not maps, they are simply booleans, no? e.g. `-setupiptables=false`

Comment (kkonnov) [reply stable/node-local-dns/values.yaml]: Yes, they are boolean. Changed. Can you check please?

Comment (kkonnov) [reply stable/node-local-dns/values.yaml]: Hi @max-rocket-internet. Any news?

Comment (max-rocket-internet) [reply stable/node-local-dns/values.yaml]: The README hasn't been updated with the new boolean values

File: stable/node-local-dns/templates/daemonset.yaml [hunk 2]
```diff
@@ -50,7 +50,15 @@ spec:
...
-        args: [ "-localip", "{{ .Values.config.localDns }},{{ .Values.config.dnsServer }}", "-conf", "/etc/Corefile", "-upstreamsvc", "{{ include "node-local-dns.fullname" . }}-upstream" ]
+        args:
+        {{- range .Values.configArgs }}
+        - {{ . }}
+        {{- end }}
+        - "{{ .Values.config.localDns }},{{ .Values.config.dnsServer }}"
+        - "{{ include "node-local-dns.fullname" . }}-upstream"
+        - -skipteardown={{ .Values.config.skipTeardown }}
+        - -setupinterface={{ .Values.config.setupInterface }}
+        - -setupiptables={{ .Values.config.setupIptables }}
```
Comment (max-rocket-internet) [inline stable/node-local-dns/templates/daemonset.yaml:61]: This is change you are re-ordering all the args and it looks like the order matters, no?

Before: `-localip 169.254.20.25,172.20.0.10 -conf /etc/Corefile -upstreamsvc node-local-dns-upstream`

After: `-localip -conf /etc/Corefile -upstreamsvc 169.254.20.25,172.20.0.10 node-local-dns-upstream...`

**Comment by kkonnov** [comment]
yeah, my bad.
In that case, I suggest this structure:
- more readable
- possibility to redefine
- independent of the order of variables

Please take a look

**Review by max-rocket-internet — APPROVED** [review]
OK, nice, thanks @kkonnov

**Pull request merged**